subtrack.MAUI/Services/SubscriptionService.cs

```diff
@@ -14,5 +15,15 @@ public async Task<IEnumerable<Subscription>> GetSubscriptions()
         {
             return await _context.Subscriptions.ToListAsync();
         }
+
+        public async Task<Subscription> GetById(int id)
```

Comment:
```suggestion
        public async Task<Subscription?> GetById(int id)
```

Instead of throwing an exception in this case just return the null subscription and let the view handle that case